modules/aruco/src/aruco.cpp

```diff
@@ -0,0 +1,1573 @@
+        accepted[i].copyTo(m);
+    }
+
+    _ids.create((int)ids.size(), 1, CV_32SC1);
+    for(unsigned int i = 0; i < ids.size(); i++)
+        _ids.getMat().ptr< int >(0)[i] = ids[i];
+
+    if(_rejected.needed()) {
+        _rejected.create((int)rejected.size(), 1, CV_32FC2);
+        for(unsigned int i = 0; i < rejected.size(); i++) {
+            _rejected.create(4, 1, CV_32FC2, i, true);
+            Mat m = _rejected.getMat(i);
+            rejected[i].copyTo(m);
+        }
+    }
+}
+
+
+/**
+  * @brief Final filter of markers after its identification
+  */
+static void _filterDetectedMarkers(InputArrayOfArrays _inCorners, InputArray _inIds,
+                                   OutputArrayOfArrays _outCorners, OutputArray _outIds) {
+
+    CV_Assert(_inCorners.total() == _inIds.total());
+    if(_inCorners.total() == 0) return;
+
+    // mark markers that will be removed
+    vector< bool > toRemove(_inCorners.total(), false);
+    bool atLeastOneRemove = false;
+
+    // remove repeated markers with same id, if one contains the other (doble border bug)
+    for(unsigned int i = 0; i < _inCorners.total() - 1; i++) {
+        for(unsigned int j = i + 1; j < _inCorners.total(); j++) {
+            if(_inIds.getMat().ptr< int >(0)[i] != _inIds.getMat().ptr< int >(0)[j]) continue;
+
+            // check if first marker is inside second
+            bool inside = true;
+            for(unsigned int p = 0; p < 4; p++) {
+                Point2f point = _inCorners.getMat(j).ptr< Point2f >(0)[p];
+                if(pointPolygonTest(_inCorners.getMat(i), point, false) < 0) {
+                    inside = false;
+                    break;
+                }
+            }
+            if(inside) {
+                toRemove[j] = true;
+                atLeastOneRemove = true;
+                continue;
+            }
+
+            // check the second marker
+            inside = true;
+            for(unsigned int p = 0; p < 4; p++) {
+                Point2f point = _inCorners.getMat(i).ptr< Point2f >(0)[p];
+                if(pointPolygonTest(_inCorners.getMat(j), point, false) < 0) {
+                    inside = false;
+                    break;
+                }
+            }
+            if(inside) {
+                toRemove[i] = true;
+                atLeastOneRemove = true;
+                continue;
+            }
+        }
+    }
+
+    // parse output
+    if(atLeastOneRemove) {
+        vector< Mat > filteredCorners;
+        vector< int > filteredIds;
+
+        for(unsigned int i = 0; i < toRemove.size(); i++) {
+            if(!toRemove[i]) {
+                filteredCorners.push_back(_inCorners.getMat(i).clone());
+                filteredIds.push_back(_inIds.getMat().ptr< int >(0)[i]);
+            }
+        }
+
+        _outIds.create((int)filteredIds.size(), 1, CV_32SC1);
+        for(unsigned int i = 0; i < filteredIds.size(); i++)
+            _outIds.getMat().ptr< int >(0)[i] = filteredIds[i];
```

Comment:
There is a way to set a Mat to a vectors memory. It s part of the Mat constructor. You wouldn't have to iterate.